chat/chat-api/api/src/java/org/sakaiproject/chat2/model/TransferableChatMessage.java

```diff
@@ -0,0 +1,95 @@
+package org.sakaiproject.chat2.model;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Date;
+
+import lombok.Data;
+
+/**
+ * 
+ * Class used to transfer data through jGroups
+ *
+ */
+@Data
+public class TransferableChatMessage implements Serializable {
+
+	public enum MessageType {
+	    CHAT,
+	    HEARTBEAT,
+	    CLEAR,
+	    REMOVE;
+	}
+	
+	public MessageType type;
+	public String id;
+	public String owner;
+	public String siteId;
+	public String channelId;
+	public String content;
+	public long timestamp;
```

Comment:
At this moment the timestamp is used locally in each cluster node. It represents the moment a message:
- is created locally
- is received via jGroups
I've been testing to send a ZonedDateTime with the moment each message is created in its own node, but if the nodes' clock are not exactly synchronized, this process produces unexpected results.